Bids validator (#144)
aramis-lab/clinica: docs/BIDS.md

├── ...
└── participants.tsv
```

## Validation of BIDS datasets

[bids-validator](https://github.com/bids-standard/bids-validator) can be run to ensure that a dataset is BIDS-compliant.

> Some errors and files can be ignored by the validator in respectively `.bids-validator-config.json` and 

Reviewer: ```suggestion
Clinica provides tools to curate several publicly available neuroimaging datasets ([ADNI](../Converters/ADNI2BIDS/), [AIBL](../Converters/AIBL2BIDS/), [NIFD](../Converters/NIFD2BIDS/) and [OASIS](../Converters/OASIS2BIDS/)) and convert them to BIDS. We decided to ignore several warnings and errors detected by the validator. These are listed in the `.bids-validator-config.json` and `.bidsignore` files at the root of each BIDS folder. 
```

> `.bidsignore` files at the root of the BIDS folder.

Reviewer: ```suggestion

```

> These files are automatically generated by clinica converters to ignore the following issues:

Reviewer: ```suggestion
These files are automatically generated by Clinica converters to ignore the following issues:
```

> - Won't fix errors:

Reviewer: It needs some background to be understood:

```suggestion
Clinica provides tools to curate several publicly available neuroimaging datasets ([ADNI](../Converters/ADNI2BIDS/), [AIBL](../Converters/AIBL2BIDS/), [NIFD](../Converters/NIFD2BIDS/) and [OASIS](../Converters/OASIS2BIDS/)) and convert them to BIDS. We decided to ignore several warnings and errors detected by the validator. These are listed in the `.bids-validator-config.json` and `.bidsignore` files at the root of each BIDS folder. These files are automatically generated by Clinica converters to ignore the following issues:

- Won't fix errors:
```

(New line missing before list)

    - PET modality is not BIDS-compliant yet, it will be after merging [BEP 009](https://github.com/bids-standard/bids-specification/pull/633)

>     (`pet/` in `.bidsignore` / `SCANS_FILENAME_NOT_MATCH_DATASET` in `.bids-validator-config.json`).

Reviewer 2: I would change the wording about it:

> PET modality is not officially supported by BIDS but is currently under revisions. See [BEP 009](https://github.com/bids-standard/bids-specification/pull/633) for details (`pet/` in `.bidsignore` / `SCANS_FILENAME_NOT_MATCH_DATASET` in `.bids-validator-config.json`). 

I am not sure that providing the JSON files for PET modality according to upcoming specifications of BIDS-PET will be straightforward to implement 🤔

Author: Done in https://github.com/aramis-lab/clinica/pull/144/commits/1e8870687de35755191595a3e75478a0be86a1a3

>     - All participants do not have the same sessions modalities (`MISSING_SESSION` / `INCONSISTENT_SUBJECTS`).

Reviewer: ```suggestion
    - All participants do not have the same sessions or modalities (`MISSING_SESSION` / `INCONSISTENT_SUBJECTS`).
```

    - JSON files with column description are not generated as TSV files already explicit the purpose of the
    columns in metadata files (`CUSTOM_COLUMN_WITHOUT_DESCRIPTION`).

>     - The optional field `authors` is no filled in `dataset_description.json` (`NO_AUTHORS`).

Reviewer: ```suggestion
    - The optional field `authors` is not filled in `dataset_description.json` (`NO_AUTHORS`).
```

>     - The folder `conversion_info` is not BIDS-compliant (`conversion_info/` in `.bidsignore`)

Reviewer: ```suggestion
    - The folder `conversion_info` is not BIDS-compliant (`conversion_info/` in `.bidsignore`).
```

> - fMRI-specific: these errors are not fixed as there are not yet preprocessing pipelines for this modality in clinica

Reviewer: ```suggestion
- fMRI-specific: these errors are not fixed as there are no preprocessing pipelines for this modality in Clinica yet
```
